### PR TITLE
Homework2: Marina Paromenskaya

### PR DESCRIPTION
**Pull request by MarinaParomenskaya**

### CODE REVIEW

File: Homework2/Hackerrank.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# of each sock, determine how many pairs of socks with
+# matching colors there are.
+
+n = 9
+ar = [10, 20, 20, 10, 10, 30, 50, 10, 20]
+pairs = 0
+for i in range(0, n):
+    if ar[i] == 0:
+        continue
+    if ar.count(ar[i]) > 1:
```
Comment (belovmd) [inline Homework2/Hackerrank.py:12]: O(n**2) complexity. 
after you'll investigate dicts you can make it O(n)

File: Homework2/Codewars.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# 1. In this kata you are required to, given a string, replace every letter
+# with its position in the alphabet.
+# If anything in the text isn't a letter, ignore it and don't return it.
+
+text = "Once upon a time..."
+alphabet = 'abcdefghijklmnopqrstuvwxyz'
```
Comment (belovmd) [inline Homework2/Codewars.py]: import string
string.ascii_lowercase will give this string

File: Homework2/Codewars.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        res += ' '
+res = res.strip()
+print(res)
+
+# 2. Given an array, find the integer that appears an odd number of times.
+# There will always be only one integer that appears an odd number of times.
+
+seq = [20, 1, -1, 2, -2, 3, 3, 5, 5, 1, 2, 4, 20, 4, -1, -2, 5]
+for i in seq:
+    if seq.count(i) % 2:
```
Comment (belovmd) [inline Homework2/Codewars.py:21]: O(n**2) complexity

File: Homework2/Codewars.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        print(i)
+        break
+
+# 3. Write a function, persistence, that takes in a positive parameter num
+# and returns its multiplicative persistence, which is the number of times
+# you must multiply the digits in num until you reach a single digit.
+
+n = 999
+numOfTimes = 0
```
Comment (belovmd) [inline Homework2/Codewars.py]: not pythonic name we use snake case

File: Homework2/HW_2.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+# 1 Напишите программу, которая считает общую цену.
+# Вводится M рублей и N копеек цена, а также количество L товара
+# Посчитайте общую цену в рублях и копейках за L товаров.
+
+request = 'Цена одной вещи 3 рубля 20 копеек, посчитать 3 предмета.'
+findTheNumbers = request.split(' ')
```
Comment (belovmd) [inline Homework2/HW_2.py]: bad naming

File: Homework2/HW_2.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+)
+
+# 3 Найти самое длинное слово в введенном предложении. Учтите
+# что в предложении есть знаки препинания
+
+proverb = 'Good, things: come; to those, who wait!'
+words = proverb.split(' ')
+maxWord = ''
+for word in words:
+    word = word.strip(',.?!:;')
```
Comment (belovmd) [inline Homework2/HW_2.py]: there are additional punctuation marks

File: Homework2/HW_2.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+# 4 Вводится строка. Требуется удалить из нее повторяющиеся символы и
+# все пробелы.
+# Например, если было введено "abc cde def", то должно быть выведено
+# "abcdef".
+
+string = 'abc cde def'
+res = ''
+for letter in string:
+    if res.find(letter) < 0 and letter != ' ':
```
Comment (belovmd) [inline Homework2/HW_2.py]: if letter not in res:

File: Homework2/HW_2.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+print("\n%s" % res)
+
+# 5 Посчитать количество строчных (маленьких)
+# и прописных (больших) букв в введенной строке. Учитывать только
+# английские буквы.
+
+string = "\nA BaD wOrKmaN aLwaYs BlaMeS hIs tOOlS"
+print(string)
+string = string.replace(' ', '')
+stringUp = string.upper()
```
Comment (belovmd) [inline Homework2/HW_2.py]: bad naming

File: Homework2/HW_2.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+count = 0
+for i in range(0, len(string)):
```
Comment (belovmd) [inline Homework2/HW_2.py]: for i in string: if i.isupper()

File: Homework2/HW_2.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+      'lowercase letters in the proverb'
+      .format(up=count, low=(len(string) - count)))
+
+# 6 Выведите n-ое число Фибоначчи, используя только временные переменные,
+# циклические операторы и условные операторы. n - вводится.
+
+num = int(input("\nInput a number: "))
+(prev, cur) = (0, 1)
+for i in range(0, num - 1):
+    (prev, cur) = (cur, prev + cur)
```
Comment (belovmd) [inline Homework2/HW_2.py]: redundant parentheses

File: Homework2/HW_2.py [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+# Вариант 1
+
+num = int(input("\nInput a number:\n"))
+currentNum = num
+capacity = 0
+while currentNum % 10:
+    capacity += 1
+    currentNum //= 10
+currentNum = num
+for i in range(0, capacity // 2):
```
Comment (belovmd) [inline Homework2/HW_2.py:87]: it can be solved with one while without number of digit's number

File: Homework2/HW_2.py [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    firstNumber = currentNum // 10 ** (capacity - 1)
+    lastNumber = currentNum % 10
```
Comment (belovmd) [inline Homework2/HW_2.py]: bad naming

**Review by belovmd — COMMENTED** [review]
Please address these issues.

File: Homework2/Hackerrank.py (hunk 1 above)
Comment (MarinaParomenskaya) [reply Homework2/Hackerrank.py:12]: OK

File: Homework2/Codewars.py (hunk 2 above)
Comment (MarinaParomenskaya) [reply Homework2/Codewars.py]: thanks

File: Homework2/Codewars.py (hunk 3 above)
Comment (MarinaParomenskaya) [reply Homework2/Codewars.py:21]: I'll try to do it with dicts in the next homework.

File: Homework2/Codewars.py (hunk 4 above)
Comment (MarinaParomenskaya) [reply Homework2/Codewars.py]: done

File: Homework2/HW_2.py (hunk 5 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: done

File: Homework2/HW_2.py (hunk 6 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: done

File: Homework2/HW_2.py (hunk 7 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: thanks

File: Homework2/HW_2.py (hunk 8 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: done

File: Homework2/HW_2.py (hunk 9 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: thanks

File: Homework2/HW_2.py (hunk 10 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: done

File: Homework2/HW_2.py (hunk 11 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py:87]: Yes, v2 is below

File: Homework2/HW_2.py (hunk 12 above)
Comment (MarinaParomenskaya) [reply Homework2/HW_2.py]: done

File: Homework2/HW_2.py [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+# 5 Посчитать количество строчных (маленьких)
+# и прописных (больших) букв в введенной строке. Учитывать только
+# английские буквы.
+
+proverb = "\nA BaD wOrKmaN aLwaYs BlaMeS hIs tOOlS"
+print(proverb)
+proverb = proverb.replace(' ', '')
+count = 0
+for i in proverb:
```
Comment (belovmd) [inline Homework2/HW_2.py:59]: 'sfdg33werag'

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**